tests/ert/unit_tests/run_models/test_experiment_serialization.py

```diff
@@ -0,0 +1,826 @@
+)
+def test_serializing_and_deserializing_ensemble_smoother(
+    baserunmodel_args: dict[str, Any],
+    initial_ensemble_args: dict[str, Any],
+    update_runmodel_args: dict[str, Any],
+) -> None:
+    runmodel = EnsembleSmoother(
+        **(baserunmodel_args | initial_ensemble_args | update_runmodel_args)
+    )
+    runmodel._storage.close()
+
+    runmodel_from_serialized = EnsembleSmoother.model_validate(
+        runmodel.model_dump() | _not_yet_serializable_args
+    )
+
+    assert runmodel_from_serialized.model_dump() == runmodel.model_dump()
+
+
+@given(
+    runmodel_args(), initial_ensemble_runmodel_strategy(), update_runmodel_strategy()
+)
+def test_serializing_and_deserializing_ensemble_information_filter(
+    baserunmodel_args: dict[str, Any],
+    initial_ensemble_args: dict[str, Any],
+    update_runmodel_args: dict[str, Any],
+) -> None:
+    runmodel = EnsembleInformationFilter(
+        **(baserunmodel_args | initial_ensemble_args | update_runmodel_args)
+    )
+    runmodel._storage.close()
+
+    runmodel_from_serialized = EnsembleInformationFilter.model_validate(
+        runmodel.model_dump() | _not_yet_serializable_args
+    )
+
+    assert runmodel_from_serialized.model_dump() == runmodel.model_dump()
+
+
+@given(
+    runmodel_args(),
+    initial_ensemble_runmodel_strategy(),
+    update_runmodel_strategy(),
+    multidass_strategy(),
+)
+def test_serializing_and_deserializing_esmda(
+    baserunmodel_args: dict[str, Any],
+    initial_ensemble_args: dict[str, Any],
+    update_runmodel_args: dict[str, Any],
+    multidass_args: dict[str, Any],
+) -> None:
+    runmodel = MultipleDataAssimilation(
+        **(
+            baserunmodel_args
+            | initial_ensemble_args
+            | update_runmodel_args
+            | multidass_args
+        )
+    )
+    runmodel._storage.close()
+
+    runmodel_from_serialized = MultipleDataAssimilation.model_validate(
+        runmodel.model_dump() | _not_yet_serializable_args
+    )
+
+    assert runmodel_from_serialized.model_dump() == runmodel.model_dump()
+
+
+def _create_and_verify_runmodel_snapshot(config, snapshot, cli_args, case):
+    runmodel = create_model(config, cli_args, queue.SimpleQueue())
+
+    # Override these to avoid user time/env-specifics in snapshots
+    runmodel.queue_config.queue_options.job_script = "fm_dispatch.py"
+    runmodel.queue_config.queue_options.activate_script = "activate"
+    runmodel.substitutions["<DATE>"] = "2000-01-01"
+
+    serialized = runmodel.model_dump_json(indent=2) + "\n"
+
+    # Trim off cwd to make snapshots general
+    cwd = str(os.getcwd())
+    serialized = serialized.replace(cwd, f"test-data/ert/{case}")
+
+    snapshot.assert_match(
+        serialized,
+        f"{Path(config.user_config_file).stem}.json",
+    )
+
+
+cases_to_test = [
+    # "heat_equation/config.ert",
+    # "poly_example/poly.ert",
+    "snake_oil/snake_oil.ert",
+    "snake_oil_field/snake_oil_field.ert",
```

Comment:
if snake_oil is almost a subset of snake_oil_field, then the latter might suffice?